placement/algo/mirrored.go

```diff
@@ -91,6 +91,21 @@ func (a mirroredAlgorithm) RemoveInstances(
 		return nil, err
 	}
 
+	// If the instances being removed are all the initializing instances in the placement.
+	// We just need to return these shards back to their sources.
+	if allInitializing(instanceIDs, p) {
```

Comment:
What happens if `instanceIDs` only contain a subset of instance ids that are initializing? What if it's a superset?

Reply:
then the reverting won't be triggered, it will go through the normal Remove path.